platform/api/aws/images.go

```diff
@@ -289,3 +365,147 @@ func registerImageParams(snapshotID, name, description string, diskBaseName stri
 		},
 	}
 }
+
+type copyImageResult struct {
+	region  string
+	imageID string
+	err     error
+}
+
+func (a *API) CopyImage(sourceImageID string, regions []string) (map[string]string, error) {
+	image, err := a.describeImage(sourceImageID)
+	if err != nil {
+		return nil, err
+	}
+
+	describeSnapshotRes, err := a.ec2.DescribeSnapshots(&ec2.DescribeSnapshotsInput{
+		SnapshotIds: []*string{image.BlockDeviceMappings[0].Ebs.SnapshotId},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("couldn't describe snapshot: %v", err)
+	}
+	snapshot := describeSnapshotRes.Snapshots[0]
+
+	var count int
+	ch := make(chan copyImageResult)
+	for _, region := range regions {
+		opts := *a.opts
+		opts.Region = region
+		aa, err := New(&opts)
+		if err != nil {
+			break
+		}
+		go aa.copyImageIn(a.opts.Region, sourceImageID,
+			*image.Name, *image.Description,
+			image.Tags, snapshot.Tags,
+			ch)
+		count += 1
+	}
+
+	amis := make(map[string]string)
+	for i := 0; i < count; i++ {
+		result := <-ch
+		if result.imageID != "" {
+			amis[result.region] = result.imageID
+		}
+		if err == nil {
+			err = result.err
+		}
+	}
+	return amis, err
+}
+
+func (a *API) copyImageIn(sourceRegion, sourceImageID, name, description string, imageTags, snapshotTags []*ec2.Tag, ch chan<- copyImageResult) {
```

Comment:
I think it's cleaner to just return `copyImageResult` here (since it will only have one result), and the result can be put on a channel by the caller.